### PR TITLE
Update and fix the reproducible builds docs.

### PR DESCRIPTION
**Pull request by mraszyk**
Update and fix the reproducible builds docs.

### CODE REVIEW

**Review by sesi200 — APPROVED** [review]

File: docs/developer-docs/build/backend/reproducible-builds.md [hunk 1]
```diff
@@ -124,70 +126,48 @@ For the build process to be deterministic:
...
...roduced different Wasm files upon the `build_path` variation and crashed upon the `time` variation). Using [ic-cdk-optimizer](https://internetcomputer.org/docs/current/developer-docs/build/languages/rust/rust-optimize) should help with avoiding the dependency on the build path. The `reprotest` docs say that "the `time` variation uses faketime which sometimes causes weird and hard-to-diagnose problems" so it might not be straightforward to enable the `time` variations for Rust builds. For Motoko builds, you could omit `--variations '+all,-build_path,-time'` and thereby trying out all supported variations. The `--rm` flag will destroy the canister after you close its shell. Finally, once inside of the canister, we launch `reprotest` in verbose mode (the `-vv` flags). You need to give it the build command you want to run as the first argument. Here, we assume that it’s `dfx build --network ic` - adjust it if you’re using a different build process. It will then run the build in two different environments. Finally, you need to tell `reprotest` which paths to compare at the end of the two builds. Here, we compare the Wasm code for all canisters, which is found in the `.dfx/ic` directory.
```
[the remark this answers is not in the document]
Comment (mraszyk) [reply docs/developer-docs/build/backend/reproducible-builds.md]: That's a very good catch! I fixed the terminology.

**Review by sesi200 — APPROVED** [review]

**Review by sesi200 — APPROVED** [review]

File: docs/developer-docs/build/backend/reproducible-builds.md [hunk 2]
```diff
@@ -124,70 +127,48 @@ For the build process to be deterministic:
...
.... You can also run it in non-privileged mode by excluding some of the variations; see the [reprotest manual](https://manpages.debian.org/stretch/reprotest/reprotest.1.en.html). We exclude the `time` variation for Rust builds (we observed that Rust builds crashed upon the `time` variation). The `reprotest` docs say that "the `time` variation uses faketime which sometimes causes weird and hard-to-diagnose problems" so it might not be straightforward to enable the `time` variations for Rust builds. For Motoko builds, you could omit `--variations '+all,-time'` and thereby trying out all supported variations. The `--rm` flag will destroy the canister after you close its shell. Finally, once inside of the canister, we launch `reprotest` in verbose mode (the `-vv` flags). You need to give it the build command you want to run as the first argument. Here, we assume that it’s `dfx build --network ic` - adjust it if you’re using a different build process. It will then run the build in two different environments. Finally, you need to tell `reprotest` which paths to compare at the end of the two builds. Here, we compare the Wasm code for all canisters, which is found in the `.dfx/ic` directory.
```
Comment (oggy-dfin) [inline docs/developer-docs/build/backend/reproducible-builds.md:164]: This is a bit worrying - I did not have problems with Rust builds when varying time. Could you share how you encountered the problem? Generally, invariance under time looks like a pretty basic desirable property to check, as it's easy to time to leak in.

Nit: "above `Dockerfile`" doesn't apply now as it's not listed above any more.

Comment (mraszyk) [reply docs/developer-docs/build/backend/reproducible-builds.md]: Filed an issue with faketime: https://github.com/wolfcw/libfaketime/issues/402

Comment (aisconnolly) [reply docs/developer-docs/build/backend/reproducible-builds.md]: Hi @oggy-dfin! Are you happy with this enough to resolve and merge?

Comment (oggy-dfin) [reply docs/developer-docs/build/backend/reproducible-builds.md]: Hi @aisconnolly - I'd like to get https://github.com/mraszyk/portal/pull/2 merged first, after that it's good from my side

Comment (mraszyk) [reply docs/developer-docs/build/backend/reproducible-builds.md]: Merged!

**Pull request merged**